gate-deck-plugins/src/main/kotlin/com/netflix/spinnaker/gate/plugins/DeckPluginService.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.gate.plugins
+
+import com.netflix.spectator.api.Registry
+import org.slf4j.LoggerFactory
+
+/**
+ * Provides Deck with means of discovering what plugins it needs to download and a standard interface for
+ * retrieving plugin assets.
+ */
+class DeckPluginService(
+  private val pluginCache: DeckPluginCache,
+  private val registry: Registry
+) {
+
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  private val assetHitsId = registry.createId("plugins.deckAssets.hits")
+  private val assetMissesId = registry.createId("plugins.deckAssets.misses")
+
+  /**
+   * Returns a list of all plugin versions that Deck should know how to load.
+   *
+   * Deck will be responsible for taking this manifest and requesting the `index.js` file inside of a plugin version
+   * and dynamically resolving any needed assets as it goes along.
+   */
+  fun getPluginsManifests(): List<DeckPluginVersion> {
+    if (!pluginCache.isCachePopulated()) {
+      throw CacheNotReadyException()
```

Comment:
I'd like this endpoint to block and wait for the manifest to be ready.  It will be a bad experience if Deck fails to load while we're in the middle of a gate deploy.